ports/stm32/dma.c

```diff
@@ -1120,6 +1120,36 @@ void dma_nohal_init(const dma_descr_t *descr, uint32_t config) {
         init->FIFOMode              // DMDIS
         | init->FIFOThreshold       // FTH
     ;
+
+    #if defined(STM32H7)
+    // Configure DMAMUX
+    uint32_t request = descr->sub_instance & DMAMUX_CxCR_DMAREQ_ID;
+    if ((dma >= DMA1_Stream0 && dma <= DMA1_Stream7) || (dma >= DMA2_Stream0 && dma <= DMA2_Stream7)) {
+        // DMA1/2 channels 0 to 7 are hardwired to DMAMUX1 channels 0-15.
+        uint32_t stream = (((uint32_t)dma & 0xff) - 16) / 24 + ((dma >= DMA2_Stream0) ? 8 : 0);
+        // Set DMAMUX channel peripheral request.
+        ((DMAMUX_Channel_TypeDef *)(((uint32_t)DMAMUX1_Channel0) + (stream * 4)))->CCR = request;
+        DMAMUX1_ChannelStatus->CFR = 1 << (stream & 0x1F);
+        // Configure DMAMUX request generator if needed.
+        if (request >= DMA_REQUEST_GENERATOR0 && request <= DMA_REQUEST_GENERATOR7) {
+            ((DMAMUX_RequestGen_TypeDef *)(((uint32_t)DMAMUX1_RequestGenerator0) + ((request - 1) * 4)))->RGCR = 0;
+            DMAMUX1_RequestGenStatus->RGCFR = 1 << (request - 1);
+        }
+    #if ENABLE_BDMA // TODO: BDMA is Not currently supported by this driver.
+    } else if (dma >= BDMA_Channel0 && dma <= BDMA_Channel7) {
```

Comment:
So this BDMA part of the code should never be enabled?

Reply:
TL;DR: No I only left this code here because it should be correct and will be useful if/when BDMA is supported, if you prefer I can remove it. 

Long answer: It can be enabled since there are no BDMA streams currently define so this path will never run, but first the handle in `dma_descr_t` must be changed to `void *` first (or use a union) because BDMA uses a different typedef...
```c
    struct _dma_descr_t {
    #if defined(STM32F4) || defined(STM32F7) || defined(STM32H7)
    void *instance; /// DMA_Stream_TypeDef or BDMA_Channel_TypeDef
 ````
This should work fine for both, in `dma_nohal_init` we can define another handle for BDMA.. This will only make it build but to actually make useful and support BDMA we also need to fix `dma_nohal_deinit` and probably `dma_nohal_start` to handle BDMA... Also there are no `dma_descr_t` streams define for BDMA we can start to add some (it's in D3 has access to all D3 memories and peripherals ex: SRAM4, SAI4 etc..) finally we need to add BDMA IRQ handlers.

Reply:
Ok, let's leave this code in then.